integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py

```diff
@@ -93,48 +93,60 @@ def __init__(
         :param **kwargs: Optional arguments that `Elasticsearch` takes.
         """
         self._hosts = hosts
-        self._client = Elasticsearch(
-            hosts,
-            headers={"user-agent": f"haystack-py-ds/{haystack_version}"},
-            **kwargs,
-        )
+        self._client = None
         self._index = index
         self._embedding_similarity_function = embedding_similarity_function
         self._custom_mapping = custom_mapping
         self._kwargs = kwargs
 
-        # Check client connection, this will raise if not connected
-        self._client.info()
-
         if self._custom_mapping and not isinstance(self._custom_mapping, Dict):
             msg = "custom_mapping must be a dictionary"
             raise ValueError(msg)
 
-        if self._custom_mapping:
-            mappings = self._custom_mapping
-        else:
-            # Configure mapping for the embedding field if none is provided
-            mappings = {
-                "properties": {
-                    "embedding": {"type": "dense_vector", "index": True, "similarity": embedding_similarity_function},
-                    "content": {"type": "text"},
-                },
-                "dynamic_templates": [
-                    {
-                        "strings": {
-                            "path_match": "*",
-                            "match_mapping_type": "string",
-                            "mapping": {
-                                "type": "keyword",
-                            },
+    @property
+    def client(self) -> Elasticsearch:
+        if self._client is None:
+            client = Elasticsearch(
+                self._hosts,
+                headers={"user-agent": f"haystack-py-ds/{haystack_version}"},
+                **self._kwargs,
+            )
+            # Check client connection, this will raise if not connected
+            client.info()
+
+            if self._custom_mapping:
+                mappings = self._custom_mapping
+            else:
```

Comment:
The only things that pops out is perhaps having `default_mappings` property and then simplifying indices creation with:

```
mappings = self._custom_mapping if self._custom_mapping else self.default_mappings
```
Makes it more obvious what's going on and easier to understand. I don't know however if this is pythonic or not. Just 2 cents.

Reply:
It does make sense indeed, just in these PRs I prefer to keep the logic changes to the bare minimum to achieve the original goal.